packages/accounts-password/password_server.js

```diff
@@ -538,44 +538,89 @@ Accounts.sendResetPasswordEmail = function (userId, email) {
   if (!email || !_.contains(_.pluck(user.emails || [], 'address'), email))
     throw new Error("No such email for user.");
 
+  var tokenReason;
+  if (reason === 'resetPassword') {
+    tokenReason = 'reset';
+  }
+  else if (reason === 'enrollAccount') {
+    tokenReason = 'enroll';
+  }
+  else {
+    throw new Error("Unknown reason: " + reason);
+  }
+
   var token = Random.secret();
   var when = new Date();
   var tokenRecord = {
     token: token,
     email: email,
     when: when,
-    reason: 'reset'
+    reason: tokenReason
   };
   Meteor.users.update(userId, {$set: {
     "services.password.reset": tokenRecord
   }});
+
   // before passing to template, update user object with new token
   Meteor._ensure(user, 'services', 'password').reset = tokenRecord;
 
-  var resetPasswordUrl = Accounts.urls.resetPassword(token);
+  var url = Accounts.urls[reason](token);
 
+  return {email, user, token, url};
+};
+
+/**
+ * @summary Creates options for email sending for reset password and enroll account emails.
+ * You can use this function when customizing a reset password or enroll account email sending.
+ * @locus Server
+ * @param {Object} email Which address of the user's to send the email to.
+ * @param {Object} user The user object to generate options for.
+ * @param {String} url URL to which user is directed to confirm the email.
+ * @param {String} reason String `resetPassword` or `enrollAccount`.
+ * @returns {Object} Options which can be passed to `Email.send`.
+ * @importFromPackage accounts-base
+ */
+Accounts.tokenEmailOptions = function (email, user, url, reason) {
```

Comment:
Again, not sure I agree on this function name.  Maybe `Accounts.generateOptionsForEmail`?  It's not really related to a `token` at all and this can be used for any account related e-mail, even one which may not have tokens, right?